nodejs/awsx/tests/ec2/ec2.spec.ts

```diff
@@ -525,34 +776,238 @@ describe("topology", () => {
 
     describe("28 block", () => {
         // a 28 block can only fit a single subnet since it only has 16 addresses available.
-        it("1 AZ, 1 subnet", () => {
-            assert.equal(topologyToJson("10.0.0.0/28", 1, [
+        it("1 AZ, 1 subnet", async () => {
+            await subnetsEqual(subnets("10.0.0.0/28", oneAZ, [
                 { type: "private" },
-            ]), `[
+            ]), [
     {
         "type": "private",
         "subnetName": "testing-private-0",
-        "availabilityZone": 0,
-        "cidrBlock": "10.0.0.0/28"
-    }
-]`);
+        args: {
+            "availabilityZone": AZ1.name,
+            "availabilityZoneId": AZ1.id,
+            "cidrBlock": "10.0.0.0/28",
+            mapPublicIpOnLaunch: false,
+            assignIpv6AddressOnCreation: false,
+        },
+    },
+]);
         });
         it("1 AZ, 2 subnets", () => {
-            assert.throws(() => topologyToJson("10.0.0.0/28", 1, [
+            assert.throws(() => subnets("10.0.0.0/28", oneAZ, [
                 { type: "public" },
                 { type: "private" },
             ]));
         });
         it("2 AZs, 1 subnets", () => {
-            assert.throws(() => topologyToJson("10.0.0.0/28", 2, [
+            assert.throws(() => subnets("10.0.0.0/28", twoAZs, [
                 { type: "private" },
             ]));
         });
         it("2 AZs, 2 subnets", () => {
-            assert.throws(() => topologyToJson("10.0.0.0/28", 2, [
+            assert.throws(() => subnets("10.0.0.0/28", twoAZs, [
                 { type: "public" },
                 { type: "private" },
             ]));
         });
     });
+
+    describe("custom locations", () => {

```

Comment:
new tests to handle people providing custom locations for subnets, and validation that we setup the subnets, natgateways and routes properly.